app/views/assignments/_assignment.html.slim

```diff
@@ -7,12 +7,13 @@ tr
   - if action_name == 'index'
     td = link_to assignment.volunteer, assignment.volunteer
   td= t("simple_form.options.assignment.state.#{assignment.state}") if assignment.state.present?
-  td= l(assignment.created_at, format: :date_only)
+  td= l(assignment.period_start) if assignment.period_start
+  td= l(assignment.period_end) if assignment.period_end
   td
     - if policy(User).show?
-      = link_to assignment.creator, assignment.creator if assignment.creator.present?
+      = link_to assignment.creator.full_name, profile_path(assignment.creator.profile) if assignment.creator.present?
```

Comment:
`if assignment.creator.present?` can actually be dropped, as an Assignment would never be valid without having one.
It could be soft deleted tho, but that case is caught by:

```ruby
belongs_to :creator, -> { with_deleted }, class_name: 'User'
```

So as long as our db is not corrupt, we can always, even if softdeleted, expect Assignment to have a creator.

Reply:
I checked in the console and if the user gets deleted, the data for the creator of the assignment are kept and the assignment is valid. So this exact check is not needed, as even if it is deleted it will return true. What could go wrong is that if the user gets deleted the link won't work.